### ISSUE TITLE
TSLint will not report unused variable on constructor parameter with modifier

### ISSUE THREAD

**Issue by litera** [palantir/tslint#1410]
### Bug Report

- __TSLint version__: TSLint for Visual Studio Code `0.5.32`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: Visual Studio Code `1.3.1`

#### TypeScript code being linted

```ts
class SomeClass {
    // TSLint will report this variable isn't used
    private unusedVar1: any;
    constructor(
        // TSLint doesn't suggest to remove "private" keyword because of unused variable
        private unusedVar2: any 
    ) {
        // doesn't use this.unusedVar2
    }
}
```

with `tslint.json` configuration:

```json
(include if relevant)
```

#### Actual behavior

TSLint doesn't report unused variable if local type-level variable is declared using accessibility modifier in constructor parameters list.

#### Expected behavior

When we add private/public/protected to a constructor parameter we also create a type-level variable.

1. If we used this specific parameter in constructor just like a normal function call parameter (without using `this`) but don't access it anywhere else using `this`, TSLint should report unused local variable.
2. If we used this specific parameter only inside a constructor but access it as local variable (using `this`) but nowhere else TSLint should suggest to not declare it as local variable but rather use it as normal function parameter (in reality to remove access modifier `private`/`public`/`protected`).
3. If we didn't use this parameter at all, it should simply report unused parameter (and _arguably_ unused local variable too).

**Comment by andy-hanson** [comment]
Should be fixed by #2235.
